app/models/setting.rb

```diff
@@ -193,11 +193,12 @@ def parse_string_value(val)
 
   # in order to avoid code duplication, this method was introduced
   def self.create_find_by_name(opts)
+    @setting_cache ||= Hash[Setting.where(:category => self.to_s).map { |s| [s.name, s] }]
```

Comment:
since we call this method from every setting subclass, e.g. https://github.com/theforeman/foreman/blob/develop/app/models/setting/general.rb#L34 via `s.update`, we load every setting separately, with this we introduce cache per setting class and load the setting from this cache. This results in single query per setting category.